### PR TITLE
Change AckReplyConsumer to expose individual methods for replying

### PR DESCRIPTION
**Pull request by davidtorres**
@garrettjonesgoogle PTAL

Note I have renamed one example file, I'm not sure if that will have unintended consequences (like in docs), please let me know.

### CODE REVIEW

File: google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/snippets/CreateSubscriptionAndConsumeMessages.java [hunk 1]
```diff
@@ -31,7 +30,7 @@
  * A snippet for Google Cloud Pub/Sub showing how to create a Pub/Sub pull subscription and
  * asynchronously pull messages from it.
  */
-public class CreateSubscriptionAndPullMessages {
+public class CreateSubscriptionAndConsumeMessages {
```
Comment (pongad) [inline google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/snippets/CreateSubscriptionAndConsumeMessages.java]: There is a reference to it in `google-cloud-pubsub/README.md`. Could you please modify there too?

File: google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/spi/v1/SubscriberImplTest.java [hunk 2]
```diff
@@ -82,13 +82,17 @@
   static class TestReceiver implements MessageReceiver {
     private final LinkedBlockingQueue<AckReplyConsumer> outstandingMessageReplies =
         new LinkedBlockingQueue<>();
-    private AckReply ackReply = AckReply.ACK;
+    private boolean shouldAck = true;
```
Comment (pongad) [inline google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/spi/v1/SubscriberImplTest.java]: Can you add a comment here? On the surface `shouldAck = false` seems to mean "don't do anything" not "nack instead" as it is the case here.

File: google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/snippets/CreateSubscriptionAndConsumeMessages.java (hunk 1 above)
Comment (davidtorres) [inline google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/snippets/CreateSubscriptionAndConsumeMessages.java]: Yes the readme is updated in this commit https://github.com/GoogleCloudPlatform/google-cloud-java/pull/1899/commits/6d6bff6158a3c746481813b0d8f2000021a785fd

File: google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/spi/v1/SubscriberImplTest.java (hunk 2 above)
Comment (davidtorres) [inline google-cloud-pubsub/src/test/java/com/google/cloud/pubsub/spi/v1/SubscriberImplTest.java]: Done

File: google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/snippets/CreateSubscriptionAndConsumeMessages.java (hunk 1 above)
Comment (garrettjonesgoogle) [inline google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/snippets/CreateSubscriptionAndConsumeMessages.java]: This still isn't done - do a search in the pubsub README for `CreateSubscriptionAndPullMessages`.

Comment (davidtorres) [inline google-cloud-examples/src/main/java/com/google/cloud/examples/pubsub/snippets/CreateSubscriptionAndConsumeMessages.java]: Ah got it, missed that one thanks.

**Comment by garrettjonesgoogle** [comment]
LGTM

**Pull request merged**